csrc/preseg_passes/pre_segmenter.cpp

```diff
@@ -36,6 +37,14 @@ namespace nvfuser::preseg_passes {
   OptimizationPass<ConsecutiveCastPass>::runPass(fusion);
   OptimizationPass<AddAxiomsPass>::runPass(fusion);
   OptimizationPass<MoveSplitCatPass>::runPass(fusion);
+  // MovePadPass needs to happen before MarkAliasPrepare and after MoveSplitCat
```

Comment:
Can you explain why? I think I know the reason but it's good to put things in the code comment for posterity.

Reply:
> after MoveSplitCat

avoid this pass moving `PadOp` around to break the `MoveSplitCat`.

> before MarkAliasPrepare

This is not necessary for the implementation as-is. i.e. the pass currently won't change anything in the graph that would impact `MarkAliasPrepare`. So having it running before or after `MarkAliasPrepare` does not matter for now.
But if we were ever to add propagation for things like broadcast/expand/squeeze, we would want `MarkAliasPrepare` to also run on the mutated graph, hence it ideally should happen before.

I will add these in the code comment